java/integTest/src/test/java/glide/BaseClientTests.java

```diff
@@ -0,0 +1,416 @@
+
+            data = client.get("set_value_with_ttl_and_update_value_with_keeping_ttl").get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void set_value_with_ttl_and_update_value_with_new_ttl() {
+        for (BaseClient client : clients) {
+            var options =
+                    SetOptions.builder()
+                            .expiry(SetOptions.TimeToLive.builder().count(100500).type(MILLISECONDS).build())
+                            .build();
+            client
+                    .set("set_value_with_ttl_and_update_value_with_new_ttl", INITIAL_VALUE, options)
+                    .get(10, SECONDS);
+            var data = client.get("set_value_with_ttl_and_update_value_with_new_ttl").get(10, SECONDS);
+            assertEquals(INITIAL_VALUE, data);
+
+            options =
+                    SetOptions.builder()
+                            .expiry(SetOptions.TimeToLive.builder().count(2000).type(MILLISECONDS).build())
+                            .build();
+            client
+                    .set("set_value_with_ttl_and_update_value_with_new_ttl", ANOTHER_VALUE, options)
+                    .get(10, SECONDS);
+            data = client.get("set_value_with_ttl_and_update_value_with_new_ttl").get(10, SECONDS);
+            assertEquals(ANOTHER_VALUE, data);
+
+            Thread.sleep(2222); // sleep a bit more than new TTL
+
+            data = client.get("set_value_with_ttl_and_update_value_with_new_ttl").get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void set_expired_value() { // expiration is in the past
+        for (BaseClient client : clients) {
+            var options =
+                    SetOptions.builder()
+                            .expiry(SetOptions.TimeToLive.builder().count(100500).type(UNIX_SECONDS).build())
+                            .build();
+            client.set("set_expired_value", INITIAL_VALUE, options).get(10, SECONDS);
+            var data = client.get("set_expired_value").get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void set_missing_value_and_returnOldValue_is_null() {
+        for (BaseClient client : clients) {
+            String ok = client.set(KEY_NAME, INITIAL_VALUE).get(10, SECONDS);
+            assertEquals(OK, ok);
+
+            var options = SetOptions.builder().returnOldValue(true).build();
+            var data = client.set("another", ANOTHER_VALUE, options).get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void decr_and_decrBy_existing_key() {
+        String key = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertEquals(OK, client.set(key, "10").get(10, SECONDS));
+
+            assertEquals(client.decr(key).get(10, SECONDS), 9);
```

Comment:
test with more than just 10 - maybe also `0` and `-10`?

Reply:
Took these from the python and node IT tests.